src/main/java/com/leverx/learningmanagementsystem/core/security/converter/CustomTokenAuthenticationConverter.java

```diff
@@ -0,0 +1,66 @@
+package com.leverx.learningmanagementsystem.core.security.converter;
+
+import com.sap.cloud.security.xsuaa.XsuaaServiceConfiguration;
+import com.sap.cloud.security.xsuaa.token.TokenAuthenticationConverter;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.authentication.AbstractAuthenticationToken;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationToken;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Objects.nonNull;
+
+@Slf4j
```

Comment:
why it's required to extend TokenAuthenticationConverter? Better implement your own by implementing Converter<Jwt, AbstractAuthenticationToken> :)